### DISCUSSION TITLE
[Command] Windows WSL2 SSH Tunnel Authentication failed PHPStorm and DBeaver

### DISCUSSION THREAD

**Discussion by AnimNyan** [wardenenv/warden#692] [Help & Support]
### Version of Warden

0.14.0

### Operating System and Installation Method

Windows 11 via WSL 2 via Homebrew

### Describe the Bug

I'm getting an error: Cannot connect to remote host: host not found from using windows using WSL2 with warden for magento 2.4.6 when trying to connect to the database with DBeaver. 

The error I'm getting is in PHPStorm when I'm setting up the SSH configuration to connect to the database from warden. I'm getting a Connection to user@tunnel.warden.test:2222 Cannot connect to remote host: host not found . Even when I press the test connection button in SSH configuration window for PHPStorm I'm getting this host not found error.

I think the error is probably because I'm using windows with WSL2 using the ubuntu VM and the extra step is causing the ssh connection to fail.

I've tried using these configuration settings to access my Database from Phpstorm and DBeaver: https://docs.warden.dev/configuration/database.html#phpstorm. My settings in PHPStorm for the SSH Configuration is Host: tunnel.warden.test , Port: 2222, Username: user, Private key file: \\wsl$\Ubuntu\home\<myusername>\.warden\tunnel\ssh_key and Parse config file ~/.ssh/config is turned on.

For the settings on the General PHPStorm page I'm using Host: youtube-demo-db-1 (this is the name of my container from warden env ps), Port: 3306, User: magento, Password: magento: Database: magento

However, when I try the command within my ubuntu WSL2 command line:
```
ssh user@tunnel.warden.test
Welcome to the Warden SSH tunnel container!

This tunnel container is used only for forwarding TCP
connections, generally to port 3306 of db containers
and is not typically used with an interactive shell.
```

I've also checked in my /etc/ssh/ssh_config and see this entry which I see was mentioned above:
```
## WARDEN START ##
Host tunnel.warden.test
HostName 127.0.0.1
User user
Port 2222
IdentityFile ~/.warden/tunnel/ssh_key
## WARDEN END ##
```

I looked at the solution here: https://github.com/wardenenv/warden/issues/27#issue-512203247 by @erikhansen but I don't see this file ~/.ssh/config in my Ubuntu WSL 2 machine.

I'm seeing this:
```
cat ~/.ssh/config
cat: /home/<myusername>/.ssh/config: No such file or directory
```

So I don't have anything to delete I'm just very confused because ssh works fine but the ssh configuration to my WSL 2 ubuntu vm running mysql does not work.

I've also tried using:
 warden svc up --force-recreate tunnel

But it doesn't work, the host not found error still happens.
As mentioned in this issue here: https://github.com/wardenenv/warden/issues/260#issuecomment-722355346

And I've also tried warden env down and warden env up to recreate all the containers 3 times and the error still happens.

but it doesn't make a difference the host is unknown in php storm for the ssh connection and it refuses to connect.

**Temporary workaround:**
using warden db connect to run sql commands from the commandline instead which works.

### To Reproduce

1. Install WSL 2 for windows.
2. Install warden for WSL 2 through homebrew.
3. Install magento 2.4.6.
4. Use PHPStorm to configure a data source with warden's recommended settings: https://docs.warden.dev/configuration/database.html#phpstorm
5. When configuring SSH Configurations in PHPStorm, press the Test Connection button.
6. See error Connection to user@tunnel.warden.test:2222 Cannot connect to remote host: host not found.

### Expected Behavior

When configuring SSH Connection in PHPStorm, press the Test Connection button using these settings: https://docs.warden.dev/configuration/database.html#phpstorm

And pressing Test Connection the SSH connection to Warden running on WSL2 should have connection successful. And PHPStorm should be able to connect to the magento database running through WSL2 on warden.

**Comment by tonybenny2020** [comment]
Hi All,
The solution for this issue is to update your JDBC drivers for MariaDB if you are using MariaDB .This can be achieved by Going to Dbeaver->Database->Driver Manager and then Double click on MariaDB in the list which will open a dialog window with tabs including Settings, Libraries etc, go to the Libraries tab and then click on Download/Update in the right-hand side. Then please download the update by checking the message 'MariaDB driver files are missing' in the dialog box. Don't forget to tick the Force download/overwrite checkbox, Click on the download botton and you will get the latest update of JDBC driver. Restart the Dbeaver and Just delete your DB connection in the Dbeaver. and re add the DB connection with the warden SSH details and connection setting by deleting the older connection settings. Click on the test connection it will work fine. Refer the screenshots for easier fix.

![1](https://github.com/user-attachments/assets/6cd4b376-8b21-482b-9a00-27ac47439847)
![2](https://github.com/user-attachments/assets/e044e943-5614-497b-941b-ac24a73fb30b)
![3](https://github.com/user-attachments/assets/41c5c6cd-8626-4c85-955a-cfbb1ddf6015)
![4](https://github.com/user-attachments/assets/b6d2a39f-4a2c-491d-97cc-70f73e43e61e)
![Screenshot 2024-08-05 091138](https://github.com/user-attachments/assets/e7d75b80-5482-45ad-9825-69771dfc74da)

**Comment by sajidunnar** [comment]
### Windows → WSL Warden DB: SSH Tunnel Fix

**Issue (symptoms)**

DbVisualizer/DBeaver on Windows failed with: “Can’t initialize tunnel / Error establishing SSHJ tunnel.”

tunnel.warden.test could not be resolved from Windows (ping/ssh failed), but worked inside WSL.

Root cause

tunnel.warden.test is only resolvable/reachable inside WSL/Warden’s network.

The GUI tried to open an SSH tunnel directly from Windows to tunnel.warden.test, which isn’t routable there.

Fix (working approach)
Create the SSH tunnel inside WSL, then point the Windows DB client to localhost.

`
# In WSL
# Open local forward (leave it running; change 3307 if needed)
ssh -o StrictHostKeyChecking=no -o ServerAliveInterval=60 -o ExitOnForwardFailure=yes \
    -N -L 3307:db:3306 user@tunnel.warden.test
`
DB client (Windows)

Host: localhost

Port: 3307
`# Windows PowerShell
Test-NetConnection localhost -Port 3307`
